pkg/apis/managed/v1alpha1/subjectpermission_types.go

```diff
@@ -7,11 +7,13 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
-// GroupPermissionSpec defines the desired state of GroupPermission
+// SubjectPermissionSpec defines the desired state of SubjectPermission
 // +k8s:openapi-gen=true
-type GroupPermissionSpec struct {
-	// Name of the Group granted permissions by the operator
-	GroupName string `json:"groupName"`
+type SubjectPermissionSpec struct {
+	// Kind of the Subject that is being granted permissions by the operator
+	SubjectKind string `json: "SubjectKind"`
```

Comment:
Rethinking it, maybe this should be just a Subject struct: https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/rbac/types.go#L66

I.e. get rid of `SubjectName string` and `SubjectKind string` and just have `Subject rbac.Subject`. Thoughts @lisa @jewzaam @sam-nguyen7 ?

Reply:
I think either way will work. What is the benefit of having a Subject struct here instead of just SubjectName and SubjectKind? @fahlmant